assets/js/components/ExecutionResults/ExecutionHeader.jsx

```diff
@@ -39,24 +45,71 @@ function ExecutionHeader({
           onSave={onFilterSave}
         />
       </div>
-      {cloudProvider === UNKNOWN_PROVIDER && (
+      {isTargetCluster(targetType) && (
+        <ClusterWarningBanner cloudProvider={target.provider} />
+      )}
+      <TargetInfoBox targetType={targetType} target={target} />
+    </>
+  );
+}
+
+function BackToTargetDetails({ targetType, targetID }) {
+  switch (targetType) {
+    case TARGET_CLUSTER:
+      return (
+        <BackButton url={`/clusters/${targetID}`}>
+          Back to Cluster Details
+        </BackButton>
+      );
+    case TARGET_HOST:
+      return (
+        <BackButton url={`/hosts/${targetID}`}>Back to Host Details</BackButton>
+      );
+    default:
+      return null;
+  }
+}
+
+function TargetInfoBox({ targetType, target }) {
+  switch (targetType) {
+    case TARGET_CLUSTER:
+      return (
+        <ClusterInfoBox haScenario={target.type} provider={target.provider} />
+      );
+    case TARGET_HOST:
+      return (
+        <HostInfoBox
+          provider={target.provider}
+          agentVersion={target.agent_version}
+        />
+      );
+    default:
+      return null;
+  }
+}
+
+function ClusterWarningBanner({ cloudProvider }) {
+  switch (cloudProvider) {
+    case UNKNOWN_PROVIDER:
+      return (
         <WarningBanner>
           The following results are valid for on-premise bare metal platforms.
           <br />
           If you are running your HANA cluster on a different platform, please
           use results with caution
         </WarningBanner>
-      )}
-      {cloudProvider === VMWARE_PROVIDER && (
+      );
+    case VMWARE_PROVIDER:
+      return (
         <WarningBanner>
           Configuration checks for HANA scale-up performance optimized clusters
           on VMware are still in experimental phase. Please use results with
           caution.
         </WarningBanner>
-      )}
-      <ClusterInfoBox haScenario={clusterScenario} provider={cloudProvider} />
-    </>
-  );
+      );
+    default:
+      return null;
+  }
 }
```

Comment:
This is an old component, your call but maybe we can have this declared into an external file

Reply:
Went for a simpler conditional rendering
```js
{targetCluster && target.provider === UNKNOWN_PROVIDER && (
  <WarningBanner>
    ...
  </WarningBanner>
)}
{targetCluster && target.provider === VMWARE_PROVIDER && (
  <WarningBanner>
    ...
  </WarningBanner>
)}
```
How does this work?